aws/resource_aws_transfer_user.go

```diff
@@ -0,0 +1,239 @@
+
+	descOpts := &transfer.DescribeUserInput{
+		UserName: aws.String(userName),
+		ServerId: aws.String(serverID),
+	}
+
+	log.Printf("[DEBUG] Describe Transfer User Option: %#v", descOpts)
+
+	resp, err := conn.DescribeUser(descOpts)
+	if err != nil {
+		if isAWSErr(err, transfer.ErrCodeResourceNotFoundException, "") {
+			log.Printf("[WARN] Transfer User (%s) for Server (%s) not found, removing from state", userName, serverID)
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("arn", resp.User.Arn)
+	d.Set("home_directory", resp.User.HomeDirectory)
+	d.Set("policy", resp.User.Policy)
+	d.Set("role", resp.User.Role)
+
+	if err := d.Set("tags", tagsToMapTransferServer(resp.User.Tags)); err != nil {
+		return fmt.Errorf("Error setting tags: %s", err)
+	}
+	return nil
+}
+
+func resourceAwsTransferUserUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).transferconn
+	updateFlag := false
+	userName := d.Get("user_name").(string)
+	serverID := d.Get("server_id").(string)
+
+	updateOpts := &transfer.UpdateUserInput{
+		UserName: aws.String(userName),
+		ServerId: aws.String(serverID),
+	}
+
+	if d.HasChange("home_directory") {
+		updateOpts.HomeDirectory = aws.String(d.Get("home_directory").(string))
+		updateFlag = true
+	}
+
+	if d.HasChange("policy") {
+		updateOpts.Policy = aws.String(d.Get("policy").(string))
+		updateFlag = true
+	}
+
+	if d.HasChange("role") {
+		updateOpts.Role = aws.String(d.Get("role").(string))
+		updateFlag = true
+	}
+
+	if updateFlag {
+		_, err := conn.UpdateUser(updateOpts)
+		if err != nil {
+			if isAWSErr(err, transfer.ErrCodeResourceNotFoundException, "") {
+				log.Printf("[WARN] Transfer User (%s) for Server (%s) not found, removing from state", userName, serverID)
+				d.SetId("")
+				return nil
+			}
+			return err
+		}
+	}
+
+	if err := setTagsTransferServer(conn, d); err != nil {
+		return fmt.Errorf("Error update tags: %s", err)
+	}
+
+	return resourceAwsTransferUserRead(d, meta)
+}
+
+func resourceAwsTransferUserDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).transferconn
+	userName := d.Get("user_name").(string)
+	serverID := d.Get("server_id").(string)
+
+	delOpts := &transfer.DeleteUserInput{
+		UserName: aws.String(userName),
+		ServerId: aws.String(serverID),
+	}
+
+	log.Printf("[DEBUG] Delete Transfer User Option: %#v", delOpts)
+
+	_, err := conn.DeleteUser(delOpts)
+	if err != nil {
+		if isAWSErr(err, transfer.ErrCodeResourceNotFoundException, "") {
+			return nil
+		}
+		return fmt.Errorf("error deleting Transfer User (%s) for Server(%s): %s", userName, serverID, err)
+	}
+
+	if err := waitForTransferUserDeletion(conn, serverID, userName); err != nil {
+		return fmt.Errorf("error waiting for Transfer User (%s) for Server (%s): %s", userName, serverID, err)
+	}
+
+	return nil
+}
+
+func resourceAwsTransferUserImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
With updating the resource ID to be more usable for import, this function should instead just handle parsing the resource ID into its constituent parts (note updated ordering to match change above) or error about the ID format:

```go
func decodeTransferUserId(id string) (string, string, error) {
	idParts := strings.SplitN(d.Id(), "/", 2)
	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
		return "", "", fmt.Errorf("unexpected format of ID (%s), expected SERVERID/USERNAME", id)
	}
	return idParts[0], idParts[1], nil
}
```

We may make a generic helper function for this in the future since its such a common pattern, but for now it can be this separate function.